src/App.tsx

```diff
@@ -1,93 +1,276 @@
+    try {
+      const receivedTodos = await getTodos(USER_ID);
+
+      setTodos(receivedTodos);
+    } catch {
+      setError('Failed to load todos');
+      setTimeout(() => {
+        setError('');
+      }, 3000);
+    } finally {
+      addTodoToProcesing(null);
+    }
+  };
+
+  const filteredTodos = useMemo(() => {
+    switch (select) {
+      case SortEnum.ACTIVE:
+        return todos.filter(({ completed }) => !completed);
+
+      case SortEnum.ALL:
+        return todos;
+
+      case SortEnum.COMPLETED:
+        return todos.filter(({ completed }) => completed);
+
+      default:
+        return todos;
+    }
+  }, [todos, select]);
+
+  useEffect(() => {
+    getTodosAll();
+  }, []);
+
+  const addTodo = async () => {
+    const newTodo: NewTodo = {
+      userId: USER_ID,
+      completed: false,
+      title: querySearch,
+    };
+
+    try {
+      await getPostTodos(USER_ID, newTodo);
+      await getTodosAll();
+    } catch {
+      setError('Unable to add a todo');
+      setTimeout(() => {
+        setError('');
+      }, 3000);
+    } finally {
+      addTodoToProcesing(null);
+    }
+  };
+
+  const handleDeleteTodo = async (todoId: number) => {
+    try {
+      await deleteTodos(todoId);
+      await getTodosAll();
+      addTodoToProcesing(todoId);
+    } catch {
+      setError('Unable to delete a todo');
+      setTimeout(() => {
+        setError('');
+      }, 3000);
+    } finally {
+      addTodoToProcesing(null);
+    }
+  };
+
+  const handleDeleteTodoCompleted = async () => {
+    todos.filter(todo => todo.completed)
+      .map(todo => handleDeleteTodo(todo.id));
+  };
+
+  const handleUpdateStatus = async (id: number) => {
+    const updatedTodo = todos.map((todo) => {
+      if (todo.id === id) {
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }
+
+      return todo;
+    });
+
+    setTodos(updatedTodo);
+    try {
+      const todoToUpdate = todos.find((todo) => todo.id === id);
+
+      if (todoToUpdate) {
+        await client.patch(`/todos/${id}`, {
+          completed: !todoToUpdate.completed,
+          title: todoToUpdate.title,
+          userId: USER_ID,
+          id,
+        });
+      }
+    } catch {
+      setError('Unable to override task status');
+      setTimeout(() => {
+        setError('');
+      }, 3000);
+    } finally {
+      addTodoToProcesing(null);
+    }
+  };
+
+  const handleUpdateAllTodoStatus = async () => {
+    const hasCompleted = todos.some(todo => !todo.completed);
+
+    const updatedAllTodo = todos.map((todo) => {
+      if (hasCompleted) {
+        return {
+          ...todo,
+          completed: true,
+        };
+      }
+
+      return {
+        ...todo,
+        completed: !todo.completed,
+      };
+    });
+
+    setTodos(updatedAllTodo);
+    try {
+      if (hasCompleted) {
+        todos.forEach(async ({
+          id,
+        }) => {
+          await client.patch(`/todos/${id}`, {
```

Comment:
'client.patch(`/todos/${id}`, {' is duplicated, please move it to the separate function